### PR TITLE
Paywall block: avoid undefined variable warning

### PR DESCRIPTION
**Pull request by simison**
@lezama can you double check the logic here, I'm not entirely sure I followed it right.

We could add some comments between lines to make it easier to follow what happens.

## Proposed changes:

Fixes:
```
PHP Warning:  Undefined variable $paywalled_content in /usr/local/src/jetpack-monorepo/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php on line 774
```
### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Post which is free _and_ paid as a free _and_ paid user (so 4 test cases)
* Same as above but with paywall block in the post
* Check both emails and the blog

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php [hunk 1]
```diff
@@ -760,15 +760,17 @@ function add_paywall( $the_content ) {
...
-			$paywalled_content = get_paywall_blocks( $post_access_level );
-		} else {
-			// emails
-			$paywalled_content = get_paywall_simple();
-		}
-		$paywalled_content = strstr( $the_content, '<!-- wp:' . $block_name . ' /-->', true ) . $paywalled_content;
+		return strstr( $the_content, '<!-- wp:' . $block_name . ' /-->', true ) . $paywalled_content;
 	}
 
 	return $paywalled_content;
```
Comment (edanzer) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:776]: This PR does seem to resolve the error for me. 

I have a question a bit beyond just resolving the error. We seem to return $paywall_content from this function whether the paywall block exists on the page or not? That was the case both before and after the changes in this PR. I would think that if there's no paywall block on the page, we should return the_content? ie:

```
if ( Jetpack_Memberships::user_can_view_post() ||  ! has_block( jetpack/paywall )) {
		return $the_content;
} else {
  if ( jetpack_is_frontend() ) {
		$paywalled_content = get_paywall_blocks( $post_access_level );
	} else {
		// emails
		$paywalled_content = get_paywall_simple();
	}
  return strstr( $the_content, '<!-- wp:' . $block_name . ' /-->', true ) . $paywalled_content;
}
```

Comment (lezama) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:776]: we return the_content earlier, here:

https://github.com/Automattic/jetpack/pull/32578/files#diff-9164d71f603706b3c6cc58c71ba3f2fbb12b1dba772adca76793f60c24ddbca7R760

Comment (edanzer) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:776]: Yup, based on Jetpack_Memberships::user_can_view_post(). 

But there's a separate check for has_block( jetpack/paywall ). So question is: what happens here if you get past the first check (ie, Jetpack_Memberships::user_can_view_post() is false), and then the post does not have a paywall block? 

In that situation, we would currently return either `get_paywall_blocks( $post_access_level )` or `get_paywall_simple()`.

Comment (lezama) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:776]: yes, that's what we want, means that the full content of the post is replaced by the paywall

Comment (edanzer) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:776]: To clarify after chatting about this: The `get_paywall_blocks( $post_access_level )` and `get_paywall_simple()` blocks are not only used to get content based on the paywall block (which I assumed based on method names) but also for broader newsletter post visibility. When `has_block( jetpack/paywall )` is false, this code will block the full content of the post based on newsletter visibility settings. When `has_block( jetpack/paywall )` is true, it'll show the content above the paywall, then block the remainder of the content.

**Review by edanzer — APPROVED** [review]

**Pull request merged**